R/validate_input.R

```diff
@@ -73,4 +86,15 @@ self_validation <- function(argument_name, class_string, len, gt0,
   stopifnot(class(must_be_integer) == "logical")
   stopifnot(length(must_be_integer) == 1)
   return(invisible(NULL))
+}
+
+validate_input2 <- function(obj, argument_name, input_set) {
```

Comment:
Do we want to add input_set as an optional argument to validate_argument? It should be possible, I guess.